dali/sasha/saarch.cpp

```diff
@@ -376,6 +376,269 @@ void WUiterate(ISashaCommand *cmd, const char *mask)
+                return false;
+
+            name.setLength(name.length()-4);
+            name.toUpperCase();
+            const char *wuid = name.str();
+            if ((name.length()>6) && strieq(wuid+name.length()-6, "_HINTS"))
+                return false;
+
+            if (!mask.isEmpty() && !WildMatch(wuid, mask.str(),true))
+                return false;
+
+            if (checkBeforeOrAfterWU)
+            {
+                if (!isEmptyString(afterWU) && (stricmp(wuid, afterWU) <= 0))
+                    return false;
+    
+                if (!isEmptyString(beforeWU) && (stricmp(wuid, beforeWU) >= 0))
+                    return false;
+                if (isEmptyString(beforeWU) || isEmptyString(afterWU))
+                    checkBeforeOrAfterWU = false; //Found the starting point. So, do not need to check anymore.
+            }
+            return true;
+        }
+        bool checkFilters(IPropertyTree *t)
+        {
+            StringBuffer path, val;
+            if (!owner.isEmpty() && (!t->getProp("@submitID", val.clear()) || !WildMatch(val.str(), owner.get(), true)))
+                return false;
+            if (!state.isEmpty() && (!t->getProp(dfu?"Progress/@state":"@state", val.clear()) || !WildMatch(val.str(), state.get(), true)))
+                return false;
+            if (!cluster.isEmpty() && (!t->getProp("@clusterName", val.clear()) || !WildMatch(val.str(), cluster.get(), true)))
+                return false;
+            if (!jobName.isEmpty() && (!t->getProp("@jobName", val.clear()) || !WildMatch(val.str(), jobName.get(), true)))
+                return false;
+            if (!cmdName.isEmpty() && (!t->getProp("@command", val.clear()) || !WildMatch(val.str(), cmdName.get(), true)))
+                return false;
+            if (!priority.isEmpty() && (!t->getProp("@priorityClass", val.clear()) || !WildMatch(val.str(), priority.get(), true)))
+                return false;
+            if (!fileRead.isEmpty() && !t->hasProp(path.setf("FilesRead/File[@name=~?\"%s\"]", fileRead.get()).str()))
+                return false;
+            if (!fileWritten.isEmpty() && !t->hasProp(path.setf("Files/File[@name=~?\"%s\"]", fileWritten.get()).str()))
+                return false;
+            if (!eclContains.isEmpty() && !t->hasProp(path.setf("Query[Text=~?\"*%s*\"]", eclContains.get()).str()))
+                return false;
+            return true;
+        }
+        void setOutput(ISashaCommand *cmd, const char *outputFormat, StringArray &selectedWUIDs,
+            IArrayOf<IPropertyTree> &selectedWUs, bool rev)
+        {
+            if (rev)
+            {
+                ForEachItemInRev(i, selectedWUIDs)
+                {
+                    StringBuffer output = selectedWUIDs.item(i);
+                    if (!isEmptyString(outputFormat))
+                        appendOutput(selectedWUs.item(i), outputFormat, output);
+                    cmd->addId(output.str());
```

Comment:
minor: if not calling appendOutput, then could pass selectedWuids.item(i) (or selectedWUs.item(i).queryName()) directly to addId and avoid the StringBuffer copy.
Also slightly more efficient to put StringBuffer outside loop to avoid deallocation and use clear() in loop.